portal/server/api.js

```diff
@@ -980,6 +987,50 @@ const getUkBankHolidays = async (token) => {
   }
 };
 
+const getCompanyByRegistrationNumber = async (registrationNumber, token) => {
+  try {
+    if (!registrationNumber) {
+      return {
+        errorMessage: 'Enter a Companies House registration number.',
+      };
+    }
+
+    if (registrationNumber && !isValidCompaniesHouseNumber(registrationNumber)) {
+      return {
+        errorMessage: 'Enter a valid Companies House registration number.',
+      };
+    }
+
+    const { data } = await axios.get(`${PORTAL_API_URL}/v1/companies/${registrationNumber}`, {
```

Comment:
I wonder if theres a structure where we could extract this into libs/common -- what do you think?